### PR TITLE
Filter against upgrading non-ASO resources

### LINKED ISSUE #3127: Bug: ASO upgrade fails on AKS when Flux is installed via AKS extensions
**Version of Azure Service Operator**
v2.1.0
```
azureserviceoperator-controller-manager   1/1     1            1           22h   manager,kube-rbac-proxy   mcr.microsoft.com/k8s/azureserviceoperator:v2.1.0,gcr.io/kubebuilder/kube-rbac-proxy:v0.13.1   control-plane=controller-manager
```

**Describe the bug**
helm upgrade fails when flux has been installed using k8s-extension. Flux installs the following CRDs:

- configsyncstatuses.clusterconfig.azure.com
- extensionconfigs.clusterconfig.azure.com
- fluxconfigs.clusterconfig.azure.com
- fluxconfigsyncstatuses.clusterconfig.azure.com

The name of these CRDs matches the pattern ASO pre-upgrade-checker looks for. As the CRDs do not set the "helm.sh/resource-policy"annotation the pre-upgrade-checker fails.

logs from pre-upgrade-checker:
```
I0706 13:11:58.691116       1 main.go:29] setup "msg"="Launching with flags" "flags"="MetricsAddr: 0, HealthAddr: , EnableLeaderElection: false, CRDPatterns: , PreUpgradeCheck: true"
E0706 13:11:59.152665       1 main.go:34] setup "msg"="pre-upgrade check failed" "error"="[CRD 'configsyncstatuses.clusterconfig.azure.com' does not have annotation for helm keep policy. Make sure the upgrade is from beta.5, CRD 'extensionconfigs.clusterconfig.azure.com' does not have annotation for helm keep policy. Make sure the upgrade is from beta.5, CRD 'fluxconfigs.clusterconfig.azure.com' does not have annotation for helm keep policy. Make sure the upgrade is from beta.5, CRD 'fluxconfigsyncstatuses.clusterconfig.azure.com' does not have annotation for helm keep policy. Make sure the upgrade is from beta.5]"
```

**To Reproduce**

- Install flux on AKS using az k8s-extension

```
 az k8s-extension create --cluster-type managedClusters --cluster-name $aksName --resource-group $resourceGroupName --name flux --extension-type microsoft.flux --configuration-settings "multiTenancy.enforce=false"
```
- Install Azure Service Operator using helm [as documented ](https://azure.github.io/azure-service-operator/#installation)
- Install succeeds
- Subsequent upgrades fail

**Expected behavior**
helm upgrade succeeds

**Screenshots**
helm upgrade logs:
image
pre-upgrade-checker logs:
image

**Additional context**
Relevant code block: https://github.com/Azure/azure-service-operator/blob/659f57df8dc411aea6a86812e524ef441b1c254d/v2/cmd/controller/app/setup.go#L69C2-L87C3

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

As reported in #3127, ASO upgrade fails when there are non-ASO CRDs present that use the `azure.com` suffix. We filter out for GVKs that we don't recognize.

Closes #3127

**How does this PR make you feel**:
gif

### CODE REVIEW

File: v2/cmd/controller/app/setup.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ package app
 import (
 	"context"
 	"fmt"
+	"github.com/Azure/azure-service-operator/v2/api"
```
Comment (matthchr) [inline v2/cmd/controller/app/setup.go:11]: minor: Ordering

**Review by matthchr — APPROVED** [review]
Approved but we should make sure to test the beta.4 -> 2.2.0 upgrade scenario before merge to ensure that this works properly.

File: v2/cmd/controller/app/setup.go [hunk 2]
```diff
@@ -74,6 +76,11 @@ func SetupPreUpgradeCheck(ctx context.Context) error {
 			continue
 		}
 
+		if !scheme.Recognizes(crd.GroupVersionKind()) {
```
Comment (super-harsh) [inline v2/cmd/controller/app/setup.go:79]: Maybe combine this with the check above? 

```
if !crdRegexp.MatchString(crd.Name) || !scheme.Recognizes(crd.GroupVersionKind()){
  continue
}
```

Comment (theunrepentantgeek) [reply v2/cmd/controller/app/setup.go:79]: I don't think that adds any clarity.

File: v2/cmd/controller/app/setup.go (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/cmd/controller/app/setup.go:11]: Fixed.

**Comment by theunrepentantgeek** [comment]
I ran a test of upgrading from earlier versions

* `beta.4` -> `v.2.2.0` (dev build) - Blocked - ✅  Pass
* `beta.4` -> `v2.0.0` - Worked - ✅  Pass
* `v2.0.0` -> `v2.2.0` (dev build) - Worked - ✅  Pass

I'll merge this PR.

**Pull request merged**